src/main/java/EventsAndCommands/ChatEventDistributor.java

```diff
@@ -0,0 +1,74 @@
+package EventsAndCommands;
+
+import java.io.DataInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import EventsAndCommands.FunEvents.GoodBotEvent;
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+
+import org.reflections.*;
+
+/**
+ * CommandHandler
+ */
+public class ChatEventDistributor extends ListenerAdapter{
+
+    private static ChatEventDistributor instance = null;
+
+    private Set<ChatEventHandler> eventHandlers;
+
+    public static ChatEventDistributor getInstance(){
+        if(instance == null){
+            instance = new ChatEventDistributor();
+        }
+        return instance;
+    }
+
+    private ChatEventDistributor(){
+        eventHandlers = new HashSet<>();
+    }
+
+    @Override
+    public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
+        for (var x : eventHandlers) {
+            if(x.trigger(event)){
+                x.receiveCommand(event);
+            }
+        }
+    }
+
+    public void registerChatEventHandler(ChatEventHandler handler){
+        eventHandlers.add(handler);
+    }
+
+    public void unregisterChatEventHandler(ChatEventHandler handler){
+        eventHandlers.remove(handler);
+    }
+
+    public void load(Class<?> c){
```

Comment:
What does this method do? It has no return type and does not add anything to a field.

Does it rely on side effects in the constructors? That sounds adventurous.